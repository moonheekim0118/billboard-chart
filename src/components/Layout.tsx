import React from 'react';
import { useSelector } from 'react-redux';

const Layout = () => {
    const number = useSelector((state) => state.test.number);

    return <div>{number}</div>;
};

export default Layout;
