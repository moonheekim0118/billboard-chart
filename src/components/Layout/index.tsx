import React from 'react';
import Navigation from '../Navigation';
import styled from 'styled-components';

interface Props {
    title: string;
    children: any;
}

const Layout = (props: Props) => {
    return (
        <>
            <Navigation />
            <Container>
                <Title>Billboard Hot 100 {props.title}</Title>
                {props.children}
            </Container>
        </>
    );
};

const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 150px;
`;

const Title = styled.span`
    font-size: 1.5em;
    font-wieght: bold;
    color: #fff;
`;

export default Layout;
