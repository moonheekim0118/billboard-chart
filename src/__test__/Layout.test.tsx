import 'jsdom-global/register';
import React from 'react';
import Layout from '../components/Layout';
import { HashRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<Layout/>', () => {
    let container = null;

    it('renders correctly', () => {
        container = mount(
            <HashRouter>
                <Layout title={'test'} children={<div>TEST</div>} />
            </HashRouter>
        );
    });

    it('matches snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});
