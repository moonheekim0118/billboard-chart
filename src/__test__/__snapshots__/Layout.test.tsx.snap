// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> matches snapshot 1`] = `"<div class=\\"sc-bdfBwQ hrWjWM\\"><div class=\\"sc-gsTCUz dACEbQ\\"><a class=\\"sc-dlfnbm ldAUts\\" href=\\"#/\\">Home</a><a class=\\"sc-dlfnbm ldAUts\\" href=\\"#/about\\">about</a></div></div><div class=\\"sc-hKgILt fIHFxu\\"><span class=\\"sc-eCssSg biEFah\\">Music test</span><div>TEST</div></div>"`;
